src/taskProvider.ts

```diff
@@ -23,6 +23,31 @@ type BuilderForLanguage = Partial<Record<TaskTargetLanguage, ExecBuilder>>;
 type TaskState = "enabled" | "disabled" | "unavailable";
 type BehaviorWhenAllTargetsSelected = "enable" | "disable" | "executeForEveryTarget";
 
+export interface ChecksList {
+  selected: string;
+  values: string[];
+}
+const validChecks: string[] = ["All", "manifest", "icons", "app_load_params", "makefile", "readme", "scan"];
+export const checks: ChecksList = {
+  selected:"All",
+  values: validChecks,
+};
+
+let checkSelectedEmitter: vscode.EventEmitter<string> = new vscode.EventEmitter();
+export const onCheckSelectedEvent: vscode.Event<string> = checkSelectedEmitter.event;
+
+export async function showChecks() {
+  let result = undefined;
+  result = await vscode.window.showQuickPick(validChecks, {
+    placeHolder: "Please select a check to run",
```

Comment:
Try to use 

```
const selected = await vscode.window.showQuickPick(items, {
            canPickMany: true, // Allows multiple selections with checkboxes
            placeHolder: 'Select one or more options'
        });
```
For allowing selecting multiple items

Reply:
Multiple selection is not allowed by the script executing the checks. it is All or a single one. Thus, no need to support multiple here, just to execute them in loop.